packages/nimble-components/src/rich-text/specs/README.md

```diff
@@ -107,6 +111,116 @@ Example usage of the `nimble-rich-text-editor` in the application layer is as fo
+        ></nimble-mapping-mention-user>
+        <nimble-mapping-mention-user
+            key="user-id-2"
+            text="Alice Smith"
+        ></nimble-mapping-mention-user>
+        <nimble-mapping-mention-user
+            key="user-id-3"
+            text="Bob Jones"
+        ></nimble-mapping-mention-user>
+    </nimble-rich-text-mention-users>
+
+    <nimble-rich-text-mention-issues>
+        <nimble-mapping-mention-user
+            key="issue-id-1"
+            text="Spec for rich text editor"
+        ></nimble-mapping-mention-user>
+        <nimble-mapping-mention-user
+            key="issue-id-2"
+            text="Mention support in rich text components"
+        ></nimble-mapping-mention-user>
+        <nimble-mapping-mention-user
+            key="issue-id-3"
+            text="Issue in pasting a link"
+        ></nimble-mapping-mention-user>
+    </nimble-rich-text-mention-issues>
+</nimble-rich-text-editor>
+```
+
+_Alternatives_:
+
+Slot based approach: All user information will be communicated through `nimble-list-option` elements. These elements will
+contain the user's ID in their `value` attribute and the user's name in their text content. When the **"@"** symbol is
+inserted into the editor, these elements will be presented as a dropdown list.
+
+Pros:
+
+1. When the client does not specify a `value` attribute, the options are sent without a user's ID, and in such cases,
+   the value attribute is derived from the content of the option.
+
+Cons:
+
+1. It is impossible to have multiple type of mentions when `nimble-list-option` is used in the default slot. If we use
+   slot names to differentiate between various type of mentions, the rich text editor needs to be aware of all the type
+   of mentions like user slot name, issue slot name etc, through an attribute value.
+2. The slot-based approach is not configurable, meaning that if a user wishes to customize the entire user mention with
+   some arbitrary value, the only viable option is to include it within the options. In contrast, the configuration-based
+   approach readily allows for the inclusion of arbitrary values in the parent container element, making customization more
+   straightforward.
+3. The UI of the list option is completely dependent on the client application.
+4. All the keyboard and mouse interactions within the dropdown list should be handled internally within the editor as `nimble-list-option`
+   does not support any.
+
+#### Client Usage Guidance on Filtered Users:
+
+Initially, the client application need not provide any user list within the children of `nimble-rich-text-editor` and it can be empty.
+The editor component will emit an event whenever the `@` character is entered into the editor. By the time, the client
+can listen to the `mention-update` event and sort the users list in alphabetical order with respect to the usernames
+and provide the initial twenty user lists to the editor via the `nimble-mapping-mention-user`.
```

Comment:
Since the header of this section is like client guidance, I have just rephrased the sentence as a recommendation instead of a client scope. I also highlighted the reason for sending the options dynamically as it will have a lot of elements in the DOM on a single page.